spec.html

```diff
@@ -37972,6 +38045,7 @@ <h1>Atomics.notify ( _typedArray_, _index_, _count_ )</h1>
         1. Let _arrayTypeName_ be _typedArray_.[[TypedArrayName]].
         1. Let _elementSize_ be the Element Size value specified in <emu-xref href="#table-the-typedarray-constructors"></emu-xref> for _arrayTypeName_.
         1. Let _indexedPosition_ be (_i_ &times; _elementSize_) + _offset_.
+        1. If IsSharedArrayBuffer(_buffer_) is *false*, return 0.
```

Comment:
Is there any particular reason that this couldn't occur as Step 2?

Reply:
I wanted to keep all the observable behavior (like the calls to ValidateAtomicAccess and ToInteger(_count_)) between the the 0-because-of-`ArrayBuffer` special case and the 0-because-no-one-is-waiting case the same.

Reply:
Ok, but that changes the order in which validation of the arguments occur. In the current spec, validation of the  `index` value will not be reached if `ValidateSharedIntegerTypedArray()` results in a thrown exception. If "return 0" replaces "throw a TypeError exception from ValidateSharedIntegerTypedArray", then "return 0" needs to be the new step 2; if it's not, then ValidateAtomicAccess is called and `index` is evaluated, which means that an additional, unintended observable behavior has been created.

```
const i64a = new BigInt64Array(
  new ArrayBuffer(BigInt64Array.BYTES_PER_ELEMENT * 8)
);

const poisonedIndex = {
  valueOf() {
    throw new Error('evaluating poisonedIndex resulted in this new observable behavior');
  }
};

// This ok... 
Atomics.notify(i64a, 0, 0) === 0;

// But this will fail because poisonedIndex is evaluated. 
Atomics.notify(i64a, poisonedIndex, 1) === 0;
```

Reply:
It was my intention to make index validation observable in symmetry with the SAB case.

Are you saying that since `Atomics.notify` is hardcoded to return 0 on ABs anyways, there's no reason it should validate the index? That's weird to me, since `Atomics.notify(someSAB, poisonedIndex, 1)` currently throws because of `poisonedIndex`, regardless of there being waiters or not. I think the simplest mental model for `Atomics.notify` on ArrayBuffers is something like "there are always zero waiters, since waiting is impossible". In that mental model, I don't see any reason to skip index validation.

Reply:
> Are you saying that since Atomics.notify is hardcoded to return 0 on ABs anyways, there's no reason it should validate the index?

Yes, that's right. 

> That's weird to me, since Atomics.notify(someSAB, poisonedIndex, 1) currently throws because of poisonedIndex,

Sure, but only because `someSAB` is a `SharedArrayBuffer`. In the current design, if it were an `ArrayBuffer`, it would fail before reaching evaluating `poisonedIndex`, and I'm suggesting that we continue to enforce the "does not reach evaluation of `index`" behavior, by placing the new special case handling of `ArrayBuffer` before the call to `ValidateAtomicAccess`.

Reply:
> Sure, but only because someSAB is a SharedArrayBuffer. In the current design, if it were an ArrayBuffer, it would fail before reaching evaluating poisonedIndex, and I'm suggesting that we continue to enforce the "does not reach evaluation of index" behavior, by placing the new special case handling of ArrayBuffer before the call to ValidateAtomicAccess.

But why? This would make it different with every other Atomics method that is relaxed by this PR to also work on ABs.

Reply:
> This would make it different from every other Atomics method that is relaxed by this PR to also work on ABs

I think I'm convinced by this: the operations that _are_ changing, will all change in a consistent way. Thanks!